### ISSUE TITLE
Low-rank Multivariate Normal distribution

### ISSUE THREAD

**Issue by fehiepsi** [pyro-ppl/numpyro#387]
This is a multivariate normal distribution with covariance matrix is parameterized by: `cov = W @ W.T + D` where `W` is a low-rank matrix, `D` is a diagonal matrix. Taking advantage of low-rank form and some algebra tricks, it is much faster to sample from or compute log probability of this distribution (comparing to MultivariateNormal distribution). If you love matrices, you might love to solve this issue. ;)

### Applications
+ This can replace MultivariateNormal priors in your model.
+ This can be used to implement a fast multivariate autoguide, which is very useful for models high dimensional latent variables.
+ Under the hood, Spare regression Gaussian Process models use this distribution as prior.

### How to implement?
+ [PyTorch implementation](https://pytorch.org/docs/stable/_modules/torch/distributions/lowrank_multivariate_normal.html) has detailed comments on the math of each method, so the best way is to mimic PyTorch implementation.
+ https://en.wikipedia.org/wiki/Woodbury_matrix_identity
+ https://en.wikipedia.org/wiki/Matrix_determinant_lemma

### What I can take away when solving this issue?
+ Embrace your linear algebra knowledge.
+ Familiarize yourself with JAX [numpy.linalg](https://jax.readthedocs.io/en/latest/jax.numpy.html#module-jax.numpy.linalg), [scipy.linalg](https://jax.readthedocs.io/en/latest/jax.scipy.html#module-jax.scipy.linalg) operators.
+ Know how to implement a distribution in NumPyro.

As usual, please comment below any related question that you have. We are happy to help you go through this issue. :)

**Comment by TuanNguyen27** [comment]
I would like to implement this! Is this the right place to put the code ? https://github.com/pyro-ppl/numpyro/blob/master/numpyro/distributions/continuous.py

**Comment by fehiepsi** [comment]
@TuanNguyen27 Yes, that's the place to put the code.

It's great to hear that you want to implement this distribution! Please let me know if you have any questions. FYI (in case you follow PyTorch implementation), we do not support the `.expand` method so you don't need to worry about `_unbroadcasted` stuff. :)
